### PR TITLE
Support more topic types when creating Event Subscription

### PR DESCRIPTION
**Pull request by EricJizbaMSFT**
- Container Registries
- Event Hubs Namespaces
- IoT Hubs

NOTE: I only list these resources - the user can't create these from VS Code.

### CODE REVIEW

File: src/eventSubscription/createWizard/ContainerRegistryListStep.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            const quickPickOptions: IAzureQuickPickOptions = { placeHolder: localize('listPlaceHolder', 'Select a registry'), id: `ContainerRegistryListStep/${wizardContext.subscriptionId}` };
+            wizardContext.registry = (await ui.showQuickPick(this.getQuickPicks(wizardContext), quickPickOptions)).data;
+        }
+
+        return wizardContext;
+    }
+
+    private async getQuickPicks(wizardContext: T): Promise<IAzureQuickPickItem<Registry>[]> {
+        const client: ContainerRegistryManagementClient = new ContainerRegistryManagementClient(wizardContext.credentials, wizardContext.subscriptionId);
+        const registries: Registry[] = await client.registries.list();
```
Comment (PrashanthCorp) [inline src/eventSubscription/createWizard/ContainerRegistryListStep.ts]: Given this line, I'm slightly cautious about the non-null assertion a few lines before. Pertaining mainly to [this issue in cosmosdb](https://github.com/Microsoft/vscode-cosmosdb/issues/628), which can be fixed with a `try-catch` similar to [this PR you pointed out](https://github.com/Microsoft/vscode-azureappservice/pull/373/files#diff-4825a11def7265b6f47fe8e35948f738R37)

Your thoughts?

Comment (EricJizbaMSFT) [reply src/eventSubscription/createWizard/ContainerRegistryListStep.ts]: Do you mean the non-null assertion a few lines _after_? The issue you referenced in Cosmos DB is for the tree view, which is much more complicated than the simple quick pick in this case. I don't think a try-catch is necessary here - worst case an invalid registry doesn't have a label, but it shouldn't throw an exception and block other registries from being listed.

File: src/eventSubscription/createWizard/EventSubscriptionCreateStep.ts [hunk 2]
```diff
@@ -32,6 +35,18 @@ export class EventSubscriptionCreateStep extends AzureWizardExecuteStep<IEventSu
...
+                    topicId = (<IContainerRegistryWizardContext>wizardContext).registry!.id!;
+                    break;
+                case TopicType.EventHubsNamespace:
+                    // tslint:disable-next-line:no-non-null-assertion
+                    topicId = (<IEventHubsNamespaceWizardContext>wizardContext).eventHubsNamespace!.id!;
+                    break;
+                case TopicType.IoTHub:
+                    // tslint:disable-next-line:no-non-null-assertion
+                    topicId = (<IIoTHubWizardContext>wizardContext).iotHub!.id!;
+                    break;
```
[the remark this answers is not in the document]
Comment (EricJizbaMSFT) [reply src/eventSubscription/createWizard/EventSubscriptionCreateStep.ts]: The non-null assertion bothers me, but it's more of a generic wizard problem and is being tracked here: https://github.com/Microsoft/vscode-azuretools/issues/159

The repetition doesn't bother me, though. I'll try any suggestions you come up with, but I'm not gonna spend time on it

File: src/eventSubscription/createWizard/EventHubsNamespaceListStep.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import { localize } from '../../utils/localize';
+
+export interface IEventHubsNamespaceWizardContext extends ISubscriptionWizardContext {
+    eventHubsNamespace?: EHNamespace;
+}
+
+export class EventHubsNamespaceListStep<T extends IEventHubsNamespaceWizardContext> extends AzureWizardPromptStep<T> {
+    public async prompt(wizardContext: T, ui: IAzureUserInput): Promise<T> {
+        if (!wizardContext.eventHubsNamespace) {
+            const quickPickOptions: IAzureQuickPickOptions = { placeHolder: localize('listPlaceHolder', 'Select an Event Hubs Namespace'), id: `EventHubsNamespaceListStep/${wizardContext.subscriptionId}` };
```
Comment (StephenWeatherford) [inline src/eventSubscription/createWizard/EventHubsNamespaceListStep.ts]: nit: "Event Hubs namespace"?

Comment (EricJizbaMSFT) [reply src/eventSubscription/createWizard/EventHubsNamespaceListStep.ts]: Lol I think it's weird, too, but that's what the portal and documentation uses
screen shot 2018-05-18 at 9 27 39 am
screen shot 2018-05-18 at 9 27 47 am

**Pull request merged**